src/App.js

```diff
@@ -77,19 +57,12 @@ const App = () => {
       name: note.name,
       description: note.description,
     };
-    const index = notesState.notes.findIndex((i) => i.id === note.id);
-    const notes = [...notesState.notes];
-    notes[index] = updatedNote;
-    setNotes({ notes });
+    const index = notesArray.findIndex((i) => i.id === note.id);
```

Comment:
It would usually be the reducer that did this. 
updateNoteAction should just take a single note